### PR TITLE
Smarter padding

### PR DESCRIPTION
**Pull request by napulen**
Zero-padding the inputs/outputs was not a good strategy.

"0" has a very special meaning for the `HarmonicRhythm7` output. The real solution to this problem is applying a `Masking` layer in the neural network, however, for implementation issues in Keras, I wasn't successful at doing this.

Instead, the padding was improved. Here is the strategy:
- Pad the inputs with a special token (`-1`), which will never appear in real inputs
- Pad each output tensor with `0`s (except in `HarmonicRhythm7`), so the network learns to map the sequences of `-1` to `0`s in those outputs
- For `HarmonicRhythm7`, pad with the last token (`6`), so that this is interpreted as "a really long time since the last chord" all throughout the padding.

### CODE REVIEW

**Comment by napulen** [comment]
Fixes #54
